### PR TITLE
Clean up following gen_kw_config refactor

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Updated documentation
- [x] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/ensemble_config.py [hunk 1]
```diff
@@ -401,21 +402,23 @@ def __repr__(self):
             + "}"
         )
 
-    def __getitem__(
-        self, key: str
-    ) -> Union[ParameterConfig, EnsembleConfig,]:
-        if key in self.py_nodes:
-            return self.py_nodes[key]
+    def __getitem__(self, key: str) -> Union[ParameterConfig, EnsembleConfig]:
```
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/ensemble_config.py:405]: This should have been ResponseConfig?

**Review by andreas-el — CHANGES_REQUESTED** [review]
Just one comment

File: src/ert/_c_wrappers/enkf/ensemble_config.py (hunk 1 above)
Comment (oyvindeide) [reply src/ert/_c_wrappers/enkf/ensemble_config.py]: Yes! Fixed now

**Review by andreas-el — APPROVED** [review]
LGTM 👍

**Pull request merged**